pandas/_libs/interval.pyi

```diff
@@ -54,8 +54,6 @@ class IntervalMixin:
     def open_left(self) -> bool: ...
     @property
     def open_right(self) -> bool: ...
-    mid: _MidDescriptor
-    length: _LengthDescriptor
```

Comment:
How about having `mid` and `length` return `Any` in `IntervalMixin` (they are defined there) and then have more concert versions in `Interval` and `IntervalTree`?

Reply:
I don't think this can work.  We had to use the `_LengthDescriptor` classes in order to have a property return different types (i.e., have an `overload` based on the type of the class inheriting from `IntervalMixin`).  So if `IntervalMixin` has `length` as a property returning `Any`, I don't think we can then make it a non-property in `Interval` or `IntervalArray` or `IntervalTree` based on these classes.

The public typing doesn't necessarily have to correspond to the implementation (which in this case is not in python!).

Having said that, your comment brought up an issue where I needed to add `mid` and `length` back to `IntervalTree` .

Reply:
Looks good to me!

A parent class seems to be allowed to have a broader type for proerties:

```py
from typing import Any

class A:
    @property
    def foo(self) -> Any:
        ...

class B(A):
    @property
    def foo(self) -> str:
        ...
```
Could have `Any` (or a union of the possible values) in `IntervalMixin`.

> The public typing doesn't necessarily have to correspond to the implementation (which in this case is not in python!).

I agree, but I think in this case we can even achieve that :)

Reply:
okay, that doesn't look type-safe

Reply:
someday this should be addressed - pyright doesn't like it when a property is in a sub-class replaced by a non-property (but we are still far away from enabling pyright's actual type checking).